includes/data-stores/class-wc-product-variation-data-store-cpt.php

```diff
@@ -0,0 +1,254 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC Variation Product Data Store: Stored in CPT.
+ *
+ * @version  2.7.0
+ * @category Class
+ * @author   WooThemes
+ */
+class WC_Product_Variation_Data_Store_CPT extends WC_Product_Data_Store_CPT implements WC_Object_Data_Store {
+
+	/*
+	|--------------------------------------------------------------------------
+	| CRUD Methods
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Reads a product from the database and sets its data to the class.
+	 *
+	 * @since 2.7.0
+	 * @param WC_Product
+	 */
+	public function read( &$product ) {
+		$product->set_defaults();
+
+		if ( ! $product->get_id() || ! ( $post_object = get_post( $product->get_id() ) ) ) {
+			return;
+		}
+
+		$id = $product->get_id();
+		$product->set_parent_id( $post_object->post_parent );
+
+		// The post doesn't have a parent id, therefore its invalid and we should prevent this being created.
+		if ( empty( $product->get_parent_id() ) ) {
+			throw new Exception( sprintf( 'No parent product set for variation #%d', $product->get_id() ), 422 );
+		}
+
+		// The post parent is not a valid variable product so we should prevent this being created.
+		if ( 'product' !== get_post_type( $product->get_parent_id() ) ) {
+			throw new Exception( sprintf( 'Invalid parent for variation #%d', $product->get_id() ), 422 );
+		}
+
+		$product->set_props( array(
+			'name'              => get_the_title( $post_object ),
+			'slug'              => $post_object->post_name,
+			'date_created'      => $post_object->post_date,
+			'date_modified'     => $post_object->post_modified,
+			'status'            => $post_object->post_status,
+			'menu_order'        => $post_object->menu_order,
+			'reviews_allowed'   => 'open' === $post_object->comment_status,
+		) );
+
+		$product = $this->read_product_data( $product );
+		$product->read_meta_data();
+		$product->set_attributes( wc_get_product_variation_attributes( $product->get_id() ) );
+
+		// Set object_read true once all data is read.
+		$product->set_object_read( true );
+	}
+
+	/**
+	 * Create a new product.
+	 *
+	 * @since 2.7.0
+	 * @param WC_Product
+	 */
+	public function create( &$product ) {
+		$product->set_date_created( current_time( 'timestamp' ) );
+
+		$id = wp_insert_post( apply_filters( 'woocommerce_new_product_variation_data', array(
+			'post_type'      => $product->post_type,
```

Comment:
I made a comment on this way above and there are a few uses of it just in case you do decide to change this.